core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -161,12 +163,7 @@ private[spark] class Executor(
    * @param interruptThread whether to interrupt the task thread
    */
   def killAllTasks(interruptThread: Boolean) : Unit = {
-    // kill all the running tasks
-    for (taskRunner <- runningTasks.values().asScala) {
-      if (taskRunner != null) {
-        taskRunner.kill(interruptThread)
-      }
-    }
+    runningTasks.keys().asScala.foreach(t => killTask(t, interruptThread = interruptThread))
```

Comment:
And the more extreme case is if `interruptThread` gets repeatedly reset, sometimes `true` and sometimes `false`.  Probably the only tractable logic will be that if `interruptThread` ever gets set to `true`, then the Task can be interrupted regardless of whether or not `interruptThread` is subsequently reset to `false`.